src/Trigger/RabbitMQListener.cs

```diff
@@ -139,12 +144,77 @@ internal void RepublishMessages(BasicDeliverEventArgs ea)
             }
         }
 
+        public string Id
+        {
+            get
+            {
+                return $"RabbitMQTrigger-{_queueName}".ToLower();
+            }
+        }
+
         private void ThrowIfDisposed()
         {
             if (_disposed)
             {
                 throw new ObjectDisposedException(null);
             }
         }
+
+        async Task<ScaleMetrics> IScaleMonitor.GetMetricsAsync()
+        {
+            return await GetMetricsAsync();
+        }
+
+        public async Task<RabbitMQTriggerMetrics> GetMetricsAsync()
+        {
+            return new RabbitMQTriggerMetrics
+            {
+                QueueLength = _queueInfo.MessageCount,
+            };
+        }
+
+        ScaleStatus IScaleMonitor.GetScaleStatus(ScaleStatusContext context)
+        {
+            ScaleStatus status = new ScaleStatus
+            {
+                Vote = ScaleVote.None,
+            };
+
+            const int NumberOfSamplesToConsider = 5;
+
+            List<RabbitMQTriggerMetrics> metrics = (context as ScaleStatusContext<RabbitMQTriggerMetrics>)?.Metrics?.ToList();
+
+            if (metrics == null || metrics.Count < NumberOfSamplesToConsider)
+            {
+                return status;
+            }
+
+            long latestQueueLength = metrics.Last().QueueLength;
+
+            if (latestQueueLength > context.WorkerCount * 1000)
+            {
+                status.Vote = ScaleVote.ScaleOut;
+                _logger.LogInformation($"QueueLength ({latestQueueLength}) > workerCount ({context.WorkerCount}) * 1000");
+                _logger.LogInformation($"Length of queue ({_queueInfo.QueueName}, {latestQueueLength}) is too high relative to the number of instances ({context.WorkerCount}).");
+                return status;
+            }
+
+            bool queueIsIdle = metrics.All(p => p.QueueLength == 0);
+
+            if (queueIsIdle)
+            {
+                status.Vote = ScaleVote.ScaleIn;
+                _logger.LogInformation($"Queue '{_queueInfo.QueueName}' is idle");
+                return status;
+            }

```

Comment:
Recommend adding the "queue length monotonically increasing" check as well, as in QueueListener